src/wp-includes/pluggable.php

```diff
@@ -1415,6 +1415,18 @@ function wp_redirect( $location, $status = 302, $x_redirect_by = 'WordPress' ) {
 
 		$location = wp_sanitize_redirect( $location );
 
+		// don't create redirect loops, but allow redirects from http to https
+		$http_host = filter_input( INPUT_SERVER, 'HTTP_HOST', FILTER_VALIDATE_DOMAIN, FILTER_FLAG_HOSTNAME );
```

Comment:
In core we don't use `filter_*` functions as the filter extension can be disabled.

Can we change this patch to not use `filter_input`?

Reply:
Yeah I don't think it's possible I'm afraid. But I just added `dev-feedback` on  the ticket to get additional thoughts on this.